### ISSUE TITLE
[Phase 4] Training Pipeline

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/seismic-event-detection#5]
## Phase 4: Training Pipeline

**Parent:** #1
**Depends on:** #4

### Objectives
Implement multi-task training with combined loss function.

### Tasks

- [ ] Implement combined loss (detection + P-pick + S-pick)
- [ ] Create training loop with progress bar
- [ ] Add validation loop
- [ ] Implement learning rate scheduling
- [ ] Add early stopping
- [ ] Save best model checkpoints
- [ ] Log training metrics

### Files to Create/Modify

| File | Action | Description |
|------|--------|-------------|
| `src/train.py` | Create | Training script |
| `configs/train_config.yaml` | Create | Training configuration |

### Code Reference

```python
def combined_loss(det_pred, det_true, p_pred, p_true, s_pred, s_true):
    """Multi-task loss for detection and phase picking."""
    # Detection: mean over time, then BCE
    det_pred_mean = det_pred.mean(dim=(1, 2))
    det_loss = F.binary_cross_entropy(det_pred_mean, det_true)

    # Pick losses: BCE per timestep
    p_pred = p_pred.squeeze(1)
    s_pred = s_pred.squeeze(1)
    p_loss = F.binary_cross_entropy(p_pred, p_true)
    s_loss = F.binary_cross_entropy(s_pred, s_true)

    return det_loss + p_loss + s_loss
```

### Training Configuration

```python
config = {
    'batch_size': 32,
    'learning_rate': 0.001,
    'epochs': 50,
    'patience': 10,
    'class_balance': 0.5,  # earthquake ratio
}
```

### Definition of Done

- [ ] Training loop runs without errors
- [ ] Loss decreases over epochs
- [ ] Validation metrics computed
- [ ] Best checkpoint saved
- [ ] Can resume from checkpoint
- [ ] GPU utilization if available

### Technical Notes

**For junior developers:**
- Multi-task learning trains all heads simultaneously
- BCE loss works for probability outputs
- Balance 50/50 earthquake/noise to avoid bias
- Monitor all three loss components separately
